platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/geometry/LatLng.java

```diff
@@ -118,6 +119,17 @@ public double getAltitude() {
         return altitude;
     }
 
+    /**
+     * Return a new LatLng object with a wrapped Longitude.  This allows original data object
+     * to remain unchanged.
+     * @return New LatLng object with wrapped Longitude
+     */
+    public LatLng wrap(){
+        LatLng wrappedVersion = new LatLng(this);
+        wrappedVersion.setLongitude(MathUtils.wrap(wrappedVersion.getLongitude(), GeoConstants.MIN_LONGITUDE, GeoConstants.MAX_LONGITUDE));
```

Comment:
I think adding the range check to avoid wrapping if we don't have to makes sense.  As for the new object part, I kinda like the idea that the `wrap()` method is consistently giving a new object so that there's never any doubt that the base object data still remains.